src/engraving/dom/tie.cpp

```diff
@@ -355,30 +355,67 @@ void Tie::calculateDirection()
     }
 }
 
-void Tie::calculateIsInside()
+void Tie::calculateIsInside(bool isUp)
 {
     if (_tiePlacement != TiePlacement::AUTO) {
         setIsInside(_tiePlacement == TiePlacement::INSIDE);
         return;
     }
 
     const Note* startN = startNote();
+    const Note* endN   = endNote();
     const Chord* startChord = startN ? startN->chord() : nullptr;
-    const Note* endN = endNote();
-    const Chord* endChord = endN ? endN->chord() : nullptr;
+    const Chord* endChord   = endN ? endN->chord() : nullptr;
+    std::vector<Note*> startNotes = startChord->notes();
+    std::vector<Note*> endNotes   = endChord->notes();
+    size_t n_start = startNotes.size();
+    size_t n_end   = endNotes.size();
```

Comment:
This will cause a crash if startChord or endChord is nullptr. Better move these four lines after the `if(!startChord || !endChord) { …; return; }` block.